src/Microsoft.Extensions.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -164,10 +173,7 @@ public void Set(string key, byte[] value, DistributedCacheEntryOptions options)
                 throw new ArgumentNullException(nameof(value));
             }
 
-            if (options == null)
```

Comment:
Keep the null check, we want a consistent contract across the IDistributedCache APIs and all the existing extensions already pass in an instance. https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Extensions.Caching.Abstractions/DistributedCacheExtensions.cs#L33